### PR TITLE
Send VG metrics as part of ThinPool metrics

### PR DESCRIPTION
**Pull request by leelavg**
fixes: #477

Signed-off-by: Leela Venkaiah G <lgangava@redhat.com>

### CODE REVIEW

**Comment by leelavg** [comment]
- I know we didn't reach consense on linked issue however providing the implementation beforehand to let know that this doesn't depend on existing PRs
- Validation after running `make -C example run BUILD_IMAGE="true"`
```
# curl -s localhost:8080/metrics | grep -Ei 'topolvm_' | grep -Ev 'HELP|TYPE'
topolvm_thinpool_data_percent{device_class="ssd-thin",node="topolvm-example-worker"} 0.55
topolvm_thinpool_metadata_percent{device_class="ssd-thin",node="topolvm-example-worker"} 11.13
topolvm_thinpool_size_bytes{device_class="ssd-thin",node="topolvm-example-worker"} 2.147483648e+09
topolvm_thinpool_vg_free_bytes{device_class="ssd-thin",node="topolvm-example-worker"} 1.8241028096e+10
topolvm_thinpool_vg_size_bytes{device_class="ssd-thin",node="topolvm-example-worker"} 2.1470642176e+10
topolvm_volumegroup_available_bytes{device_class="ssd",node="topolvm-example-worker"} 1.8241028096e+10
topolvm_volumegroup_size_bytes{device_class="ssd",node="topolvm-example-worker"} 2.1470642176e+10
```

**Review by nbalacha — COMMENTED** [review]
lgtm

**Comment by leelavg** [comment]
```
# curl -s localhost:8080/metrics | grep -Ei 'topolvm_' | grep -Ev 'HELP|TYPE'
topolvm_thinpool_data_percent{device_class="ssd-thin",node="topolvm-example-worker"} 0.55
topolvm_thinpool_metadata_percent{device_class="ssd-thin",node="topolvm-example-worker"} 11.13
topolvm_thinpool_size_bytes{device_class="ssd-thin",node="topolvm-example-worker"} 2.147483648e+09
topolvm_volumegroup_available_bytes{device_class="ssd",node="topolvm-example-worker"} 1.8241028096e+10
topolvm_volumegroup_available_bytes{device_class="ssd-thin",node="topolvm-example-worker"} 1.8241028096e+10
topolvm_volumegroup_size_bytes{device_class="ssd",node="topolvm-example-worker"} 2.1470642176e+10
topolvm_volumegroup_size_bytes{device_class="ssd-thin",node="topolvm-example-worker"} 2.1470642176e+10

# k get no -ojson|jq -r '.items[].metadata.annotations | with_entries(if (.key|test("topolvm")) then ({key: .key, value: .value }) else empty end)'
{}
{
  "capacity.topolvm.cybozu.com/00default": "18241028096",
  "capacity.topolvm.cybozu.com/ssd": "18241028096",
  "capacity.topolvm.cybozu.com/ssd-thin": "10737418240"
}```

File: docs/lvmd-protocol.md [hunk 1]
```diff
@@ -205,6 +205,7 @@ Represents the details of thinpool.
 | data_percent | [double](#double) |  | Data percent occupied on the thinpool, used for monitoring. |
 | metadata_percent | [double](#double) |  | Metadata percent occupied on the thinpool, used for monitoring. |
 | overprovision_bytes | [uint64](#uint64) |  | Free space on the thinpool with overprovision, used for annotating node. |
+| size_bytes | [uint64](#uint64) |  | Size of the thinpool. |
```
Comment (llamerada-jp) [inline docs/lvmd-protocol.md:208]: According to man page. `Physical data space size of the thinpool.` looks explicitly. Could you please consider this?
https://man7.org/linux/man-pages/man7/lvmthin.7.html

Comment (leelavg) [reply docs/lvmd-protocol.md]: sounds reasonable, done.

**Review by llamerada-jp — APPROVED** [review]
LGTM

**Review by cupnes — APPROVED** [review]
LGTM

**Pull request merged**